src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets

```diff
@@ -78,6 +78,18 @@ Copyright (c) .NET Foundation. All rights reserved.
                                           $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), '5.0'))">.NETCoreApp</TargetFrameworkIdentifier>
   </PropertyGroup>
 
+  <!--
+    Parse TargetFramework containing TargetPlatform properties.
+  -->
+  <PropertyGroup Condition="'$(TargetFramework)' != '' and $(TargetFramework.Contains('-')) and ('$(TargetFrameworkIdentifier)' == '' or '$(TargetFrameworkVersion)' == '')">
+    <TargetFrameworkIdentifier>$([MSBuild]::GetTargetFrameworkIdentifier('$(TargetFramework)'))</TargetFrameworkIdentifier>
```

Comment:
Don't need to move the existing logic to use the intrinsic function? Or we already did?

If so parsing TargetFrameworkIdentifier and TargetFrameworkVersion might be duplicated

Reply:
Right now if we try to specify a target framework of this format (ex "net5.0-windows") it errors out [here](https://github.com/dotnet/sdk/pull/12202/files#diff-b1cca2205c228925d515de229057c7c6L99) because [TargetFrameworkIdentifier or TargetFrameworkVersion is empty](https://github.com/dotnet/sdk/pull/12202/files#diff-b1cca2205c228925d515de229057c7c6L87-L89), so I think we should be good to leave the old logic without having duplication.

Reply:
Just pushed changes, I removed the old logic for parsing TargetFrameworkIdentifier and TargetFrameworkVersion and replaced it with calls to the new nuget functions.